src/io/MemoryProfiler.cpp

```diff
@@ -7,42 +7,38 @@
 
 #include "MemoryProfiler.h"
 
-#include <iomanip>
-#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <memory>
+#include <regex>
 #include <sstream>
 #include <string>
-#include <cstring>
-#include "sys/types.h"
 #include "sys/sysinfo.h"
-
+#ifndef _SX
```

Comment:
What is the preprocessor macro `_SX`? I cannot find any definition in our code base or online.

Reply:
This preprocessor macro is specific to NEC platforms which do not run standard unix/linux like the https://en.wikipedia.org/wiki/NEC_SX-Aurora_TSUBASA.

Reply:
Thank you for the clarification 👍  Does anybody still use _ls1_ on those machines or is it just legacy?

Reply:
We have at least a test system here until 31. Dec 2024 ;)